packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_request_wrapper.dart

```diff
@@ -0,0 +1,117 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:flutter/foundation.dart';
+import 'package:in_app_purchase/src/in_app_purchase_connection/product.dart';
+import '../channel.dart';
+
+
+/// [SKProductRequestWrapper] wraps IOS SKProductRequest class to to retrive StoreKit product information in dart.
+///
+/// https://developer.apple.com/documentation/storekit/skproductsrequest?language=objc
+class SKProductRequestWrapper {
+
+  static MethodChannel _channel = Channel.instance;
+  /// Get product list.
+  ///
+  /// [identifiers] is the product identifiers specified in Itunes Connect for the products that need to be retrived.
+  ///
+  /// Returns a future containing a list of [SKProduct] which then can be queried to get desired information.
+  static Future<List<Product>> getProductList(List<String> identifiers) async {
+    return _channel.invokeListMethod<Map<dynamic, dynamic>>(
+      'getProductList',
+      <String, Object>{
+        'identifiers': identifiers,
+      },
+    ).then<List<Product>>((List<Map<dynamic, dynamic>> productListJson) {
+      final List<Product> productList = <Product>[];
+      for (Map<dynamic, dynamic> productJson in productListJson) {
+        productList.add(Product(
+          skProduct: SKProductWrapper.fromJson(productJson),
+        ));
+      }
+      return productList;
+    });
+  }
+}
+
+/// https://developer.apple.com/documentation/storekit/skproduct/2936884-subscriptionperiod?language=objc
+class SKProductSubscriptionPeriodWrapper {
+  SKProductSubscriptionPeriodWrapper(
+      {@required this.numberOfUnits, @required this.unit});
+
+  SKProductSubscriptionPeriodWrapper.fromJson(Map<dynamic, dynamic> json)
```

Comment:
nit: I know this causes weird casting problems that bubble upstream, but the method signature should probably be `Map<String, dynamic>` for the type. The keys of the map all need to be strings and not variable, even though the values are dynamic.